compiler/modules/front/parseStmt.bal

```diff
@@ -147,6 +155,23 @@ function finishAssignStmt(Tokenizer tok, LExpr lValue) returns AssignStmt|err:Sy
     check tok.expect(";");
     return stmt; 
 }
+function parseArithmeticCompAssnStmt(Tokenizer tok, LExpr lValue, BinaryArithmeticOp op) returns AssignStmt|err:Syntax {
+    check tok.advance();
+    Expr rexpr = check parseExpr(tok);
+    Expr expr = {arithmeticOp: op, left: lValue, right: rexpr, pos: tok.currentPos()};
```

Comment:
This approach won't work because in `a[foo()] += 1;`, `foo` needs to be called exactly once.

You need to create a CompoundAssignStmt in the AST.